src/scenario/scenario_events_controller.c

```diff
@@ -0,0 +1,263 @@
+#include "scenario_events_controller.h"
+
+#include "core/log.h"
+#include "game/save_version.h"
+#include "scenario/action_types/action_handler.h"
+#include "scenario/condition_types/condition_handler.h"
+#include "scenario/scenario.h"
+#include "scenario/scenario_event.h"
+
+#define SCENARIO_EVENTS_SIZE_STEP 50
+
+static array(scenario_event_t) scenario_events;
+
+static int event_in_use(const scenario_event_t *event)
+{
+    return event->state != EVENT_STATE_UNDEFINED;
+}
+
+void scenario_events_init(void)
+{
+    scenario_event_t *current;
+    array_foreach(scenario_events, current) {
+        scenario_event_init(current);
+    }
+}
+
+void new_scenario_event(scenario_event_t *obj, int position)
+{
+    obj->id = position;
+}
+
+void scenario_events_clear(void)
+{
+    scenario_events.size = 0;
+    if (!array_init(scenario_events, SCENARIO_EVENTS_SIZE_STEP, new_scenario_event, event_in_use)) {
+        log_error("Unable to allocate enough memory for the scenario events array. The game will now crash.", 0, 0);
+    }
+}
+
+scenario_event_t *scenario_event_get(int event_id)
+{
+    return array_item(scenario_events, event_id);
+}
+
+scenario_event_t *scenario_event_create(int repeat_min, int repeat_max, int max_repeats)
+{
+    if (repeat_min < 0) {
+        log_error("Event minimum repeat is less than 0.", 0, 0);
+        return 0;
+    }
+    if (repeat_max < 0) {
+        log_error("Event maximum repeat is less than 0.", 0, 0);
+        return 0;
+    }
+
+    if (repeat_max < repeat_min) {
+        log_error("Event maximum repeat is less than its minimum. Swapping the two values.", 0, 0);
```

Comment:
```suggestion
        log_info("Event maximum repeat is less than its minimum. Swapping the two values.", 0, 0);
```

Reply:
I was clearly not paying close attention at this point. Apologies.
Fixed.